src/libraries/System.IO.FileSystem/tests/Base/BaseGetSetTimes.cs

```diff
@@ -71,6 +80,95 @@ public void SettingUpdatesProperties()
             });
         }
 
+        [Fact]
+        public void SettingUpdatesProperties()
+        {
+            T item = GetExistingItem();
+            SettingUpdatesPropertiesCore(item);
+        }
+
+        private void SettingUpdatesPropertiesOnSymlinkCore(bool isRelative, bool targetExists)
+        {
+            if (!targetExists && IsDirectory && !OperatingSystem.IsWindows())
+            {
+                // On Unix, symlinks that are created as 'directories' need their
+                // directory target to exist for it to count as a directory symlink.
+                return;
+            }
+
+            T target = targetExists ? GetExistingItem() : GetMissingItem();
+
+            // When the target exists, we want to verify that its times don't change.
+            // If the target doesn't exist, we just run the symlink tests, otherwise
+            // we run it in an Assert.All so it is easy to tell if it failed by not
+            // setting it on the symlink, or if it failed by setting it on the target.
+            // The symlink test is [0] of the Assert.All, and the target check is [1].
+
+            T item = CreateSymlinkToItem(target, isRelative);
+            if (!targetExists) SettingUpdatesPropertiesCore(item);
```

Comment:
> It would be good to add some randomization here. One setter could cause subsequent getters to succeed even if their setter is broken.

Are you referring to randomize the date that is set in `SettingUpdatesPropertiesCore`, or randomize the dates of the target, or both?

Reply:
```suggestion
            if (!targetExists) 
            {
                SettingUpdatesPropertiesCore(item);
            }
```

Reply:
This is in the next commit.

Reply:
> Are you referring to randomize the date that is set in `SettingUpdatesPropertiesCore`, or randomize the dates of the target, or both?

I think they're referring to randomising both, but it's unnecessary since you said path names are randomised.

Reply:
> Are you referring to randomize the date that is set in SettingUpdatesPropertiesCore, or randomize the dates of the target, or both?

`SettingUpdatesPropertiesCore`. The getters/setters run on the same path, so the path being unique doesn't detect issues where one setter affects multiple getters.

Reply:
> `SettingUpdatesPropertiesCore`. The getters/setters run on the same path, so the path being unique doesn't detect issues where one setter affects multiple getters.

There's a different test for that (for normal files). It's assumed that it will also work on symlinks if times are set correctly on symlinks rather than the targets (which this test checks), since there is no code that currently, and shouldn't ever probably be any time setting/getting code that treats symlinks differently to normal files (it would likely fail this test also if it did anyway). Note the condition which is most likely for one setter affecting multiple getters is "when the modification time is less than the creation time" (if code was done from scratch for macOS without reading some of the comments would be when this is an issue), so simply testing if particular setters change more than 1 getter isn't good enough.
I do think it's probably unnecessary, but if @Jozkee wants, I could also implement the `SettingUpdatesPropertiesAfterAnother` test for symlinks, but again, I think it's probably overkill since it's being checked for normal files.

Reply:
My comment is not specific to this PR. It applies also to the `SettingUpdatesProperties` test as it was.

Reply:
See the `SettingUpdatesPropertiesAfterAnother` test.

Reply:
Ah, yes, `SettingUpdatesPropertiesAfterAnother` will catch it.